junit5/src/main/java/com/linecorp/armeria/testing/junit5/client/HttpHeadersAssert.java

```diff
@@ -0,0 +1,193 @@
+/*
+ * Copyright 2023 LINE Corporation
+ *
+ * LINE Corporation licenses this file to you under the Apache License,
+ * version 2.0 (the "License"); you may not use this file except in compliance
+ * with the License. You may obtain a copy of the License at:
+ *
+ *   https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations
+ * under the License.
+ */
+
+package com.linecorp.armeria.testing.junit5.client;
+
+import static java.util.Objects.requireNonNull;
+import static org.junit.jupiter.api.Assertions.assertEquals;
+import static org.junit.jupiter.api.Assertions.assertFalse;
+import static org.junit.jupiter.api.Assertions.assertTrue;
+
+import com.linecorp.armeria.common.ContentDisposition;
+import com.linecorp.armeria.common.HttpHeaders;
+import com.linecorp.armeria.common.MediaType;
+
+import io.netty.util.AsciiString;
+
+/**
+ * Assertion methods for HttpData.
+ */
+public final class HttpHeadersAssert extends AbstractResponseAssert<HttpHeaders> {
+    HttpHeadersAssert(HttpHeaders actual, TestHttpResponse response) {
+        super(actual, response);
+    }
+
+    /**
+     * Asserts that the content length of actual {@link HttpHeaders} is equal to the given one.
+     */
+    public TestHttpResponse contentLengthIsEqualTo(long expected) {
+        assertEquals(expected, actual().contentLength());
+        return response();
+    }
+
+    /**
+     * Asserts that the content type of actual {@link HttpHeaders} is equal to the given one.
+     * The {@code expected} cannot be null.
+     */
+    public TestHttpResponse contentTypeIsEqualTo(MediaType expected) {
+        requireNonNull(expected, "expected");
+        assertEquals(expected, actual().contentType());
+        return response();
+    }
+
+    /**
+     * Asserts that the content disposition of actual {@link HttpHeaders} is equal to the given one.
+     * The {@code expected} cannot be null.
+     */
+    public TestHttpResponse contentDispositionIsEqualTo(ContentDisposition expected) {
+        requireNonNull(expected, "expected");
+        assertEquals(expected, actual().contentDisposition());
+        return response();
+    }
+
+    /**
+     * Asserts that the actual {@link HttpHeaders} contains the given name.
+     * The {@code name} cannot be null.
+     */
+    public TestHttpResponse contains(CharSequence name) {
+        requireNonNull(name, "name");
+        assertTrue(actual().contains(name));
+        return response();
+    }
+
+    /**
+     * Asserts that the actual {@link HttpHeaders} contains the given name and {@link String} value.
+     * The {@code name} and the {@code value} cannot be null.
+     */
+    public TestHttpResponse contains(CharSequence name, String value) {
```

Comment:
Suggestion: What do you think of

- Removing the other `contains*` variants since they don't seem to be providing much value - the object is string-ified internally anyways
- Accepting `Object` for the general case

```suggestion
    public TestHttpResponse contains(CharSequence name, Object value) {
```